tools/build-tools/README.md

```diff
@@ -13,30 +19,66 @@ In this directory:
+You can globally install this package to run them.
+To run the version of them you built in "Setup" above, run `node bin/tool-name`.
+Note that this correctly detects the fluid-framework directory when from elsewhere, for example it works just fine in the repository root with `node tools/build-tools/bin/fluid-build fluid-build`.
+
+One of these is tools is `fluid-build`:
+
+<!-- this list of arguments is duplicated in `src/fluidBuild/options.ts` and they should be updated together -->
+
 ```txt
-Usage: fluid-build <options> [<npm script>] [<package regexp> ...]
-  [<npm script>]         Name of the npm script to run (default: build)
-  [<package regexp> ...] Regexp to match the package name (default: all packages)
+Usage: fluid-build <options> [(<package regexp>|<path>) ...]
+    [<package regexp> ...] Regexp to match the package name (default: all packages)
 Options:
-     --all            Operate on all packages/monorepo (default: client monorepo)
+     --all            Operate on all packages/monorepo (default: client monorepo). See also `--server`.
   -c --clean          Same as running build script 'clean' on matched packages (all if package regexp is not specified)
   -d --dep            Apply actions (clean/force/rebuild) to matched packages and their dependent packages
+     --fix            Auto fix warning from package check if possible
   -f --force          Force build and ignore dependency check on matched packages (all if package regexp is not specified)
   -? --help           Print this message
-     --install        Run npm install for all packages/monorepo
+     --install        Run npm install for all packages/monorepo. This skips a package if node_modules already exists: it can not be used to update in response to changes to the package.json.
   -r --rebuild        Clean and build on matched packages (all if package regexp is not specified)
-     --reinstall      Same as --uninstall --install
+     --reinstall      Same as --uninstall --install.
      --root <path>    Root directory of the Fluid repo (default: env _FLUID_ROOT_)
   -s --script <name>  npm script to execute (default:build)
-     --server         Operate on the server monorepo
-     --symlink        Fix symlink between packages within monorepo (isolate mode)
-     --symlink:full   Fix symlink between packages across monorepo (full mode)
-     --uninstall      Clean all node_modules
+     --server         Operate on the server monorepo (default: client monorepo). Overridden by `--all`
+     --symlink        Fix symlink between packages within monorepo (isolate mode). This configures the symlinks to only connect within each lerna managed group of packages. This is the configuration tested by CI and should be kept working.
+     --symlink:full   Fix symlink between packages across monorepo (full mode). This symlinks everything in the repo together. CI does not ensure this configuration is functional, so it may or may not work.
```

Comment:
Probably useful to note this is primarily used for local testing of changes across lerna groups?  Also I'm pretty sure this will only link packages that have been specified to use their prereleases in the downstream dependencies which would be worth noting as well

Reply:
docs/README.md says to use symlink:full, as well as a couple of PropertyDDs packages in their readme and https://github.com/microsoft/FluidFramework/wiki/Dev-Tools-Maintenance . Its unclear why any of these say to use smylink:full, but I don't think the reason they recommend it is "for local testing of changes across lerna groups".

I don't know what symlink:full does (I thought I did, but I was wrong), or why things use it, I just know documentation says to do it sometimes, and doing it makes the build fail.